src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -184,24 +185,26 @@ public Command testTurretCommand(double degrees) {
   }
 
   private void aimTurret() {
-    double limelightTX = LimelightHelpers.getTX("limelight");
-    if (limelightTX != 0
-        && LimelightHelpers.getFiducialID("limelight")
-            == ((robotStateManager.getAllianceColor() == AllianceColor.BLUE)
-                ? Constants.TurretConstants.SPEAKER_TAG_ID_BLUE
-                : Constants.TurretConstants.SPEAKER_TAG_ID_RED)) {
-      // X & Rotation
-      setTurretPos(Math.toRadians(limelightTX) + turretPosition);
-
-      // Y & Tilting
-      double limelightTY = LimelightHelpers.getTY("limelight");
-      double distanceToTag = tyToDistanceFromTag(limelightTY);
-      tagDistanceEntry.log(distanceToTag);
-      // TODO: Add vertical tilt and use distance for it
-
-      if (Math.abs(Math.toRadians(limelightTX) + turretPosition)
-          > Math.toRadians(Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES)) {
-        // TODO: Make turret rotate the drivebase if necessary and driver thinks it's a good idea
+    if (visionSubsystem != null) {
+      int tagID =
+          ((robotStateManager.getAllianceColor() == AllianceColor.BLUE)
+              ? Constants.TurretConstants.SPEAKER_TAG_ID_BLUE
+              : Constants.TurretConstants.SPEAKER_TAG_ID_RED);
+      double limelightTX = visionSubsystem.getTurretYaw(tagID);
```

Comment:
Since you're converting to using an interface instead of using limelight directly, I'd recommend calling things `vision*` instead of `limelight*`.

Reply:
I changed them, it was just a leftover piece of code from the old version using only limelight